text/3533-combine-infra-and-release-teams.md

```diff
@@ -0,0 +1,57 @@
+- Feature Name: combine-infra-and-release
+- Start Date: 2023-11-23
+- RFC PR: [rust-lang/rfcs#3533](https://github.com/rust-lang/rfcs/pull/3533)
+- Rust Issue: N/A
+
+# Summary
+
+This RFC proposes merging the Release team into the Infrastructure team as a subteam. The membership of the Infrastructure and Release teams proper remain the same.[^subteam]
+
+[^subteam]: Note: Members of subteams are not automatically members of their parent team. So Release members will be part of the Infra team family but *not* members of the team proper.
+
+# Motivation
+
+Historically the Release team has had a much smaller scope than other teams. It typically only functions to execute on releases. While this is not a trivial amount of work, it is definitely much smaller than other top-level teams' purviews (e.g., the compiler team's purview to develop the Rust compiler).
+
+[RFC 3392](https://github.com/rust-lang/rfcs/blob/master/text/3392-leadership-council.md#top-level-teams) outlines what typically qualifies a team as "top-level". While one could make the argument that the Release team fits these points, there are arguably two aspects where it does not neatly fit:
+* "Have a purview that is foundational to the Rust Project": while Rust releases are obviously extremely important, it is hard to argue that they are "foundational". Just like there is no CI team (despite CI also being extremely important), the release process is not a foundational piece of what makes Rust, Rust.
+* "Have a purview that not is a subset of another team's purview": this is hard to argue exactly as most teams don't have well defined purviews, but one could argue that the Infrastructure team's purview is roughly "to maintain and administer all logistical activities for the continuing operation of the Rust project" which would then be a superset of what the Release team's purview is.
+
+In the past, whether a team is "top-level" or not has not been of huge consequence. However, this is no longer true since [RFC 3392](https://github.com/rust-lang/rfcs/pull/3392) introduced the Leadership Council whose representation is based on top-level team status. This RFC specifically called out the need for re-examination of which teams are top-level, and this proposal is the first attempt at such a re-examination. The most immediate practical reason for being strict with the definition of "top-level" is to balance "the desire for a diverse and relatively shallow structure while still being practical for productive conversation and consent."
+
+With this proposal the Infrastructure team's purview does not grow considerably and remains at a level similar to other existing top-level teams such as a language and compiler.
```

Comment:
`T-release` decides if there are enough intensives (PRs, critical issues) to warrant a dot release, this would IMO greatly increase `T-infra` purview, as it would make `T-infra` Users aware, i.e. take decisions for the benefit or at the expense of Rust users instead of just the project currently.

This is not necessarily a blocker, but I think it should be mentioned in the RFC.

Reply:
The infra team's purview is already (partially) directed towards users, between maintaining the download infrastructure for crates/releases, and developing and maintaining the playground.

Reply:
Added a suggestion for possible text (no strong feelings on specifics), but I broadly agree with @pietroalbini here. The particular question of whether to issue a dot release is one that at least IMO we don't have an amazing story around today. Long-term, my hope/expectation is to make that much closer to how we issue nightlies or betas (i.e., automatic on new commits being available).

But ultimately in the short term I think the expanded top-level scope, if it is expanded, is only to a pretty minor extent.

Reply:
I've added some explanation of this similar to @Mark-Simulacrum's suggestion.